### PR TITLE
Update INSTALL

### PR DESCRIPTION
**Pull request by antsankov**
This PR updates INSTALL with the correct link to the documentation on building.

### CODE REVIEW

**Comment by antsankov** [comment]
The old link no longer works, I found relevant documentation on the new link.

File: INSTALL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Building Zcash
 
-See the Zcash github wiki (https://github.com/zcash/zcash/wiki) for instructions on building zcashd,
+See the Zcash documenation wiki (https://zcash.readthedocs.io/en/latest/rtd_pages/user_guide.html) for instructions on building zcashd,
```
Comment (daira) [inline INSTALL]: Typo: should be "documentation"

Comment (antsankov) [reply INSTALL]: Fixed Typo

**Comment by Eirik0** [comment]
@zkbot r+

**Comment by Eirik0** [comment]
@zkbot retry

**Pull request merged**